tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -1229,8 +1229,20 @@ void ASTProducer::enqueueConsumer(
           Mgr->Impl.ASTCache.set(This->InvokRef->Impl.Key, This);
         }
       };
+
+      llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FS = FileSystem;
+      const InvocationOptions &InvocOpts = This->InvokRef->Impl.Opts;
+      const CompilerInvocation &ActualInvoc = InvocOpts.Invok;
+      auto ExpectedOverlay =
+          ActualInvoc.getSearchPathOptions().makeOverlayFileSystem(FileSystem);
+      if (ExpectedOverlay) {
+        FS = ExpectedOverlay.get();
+      } else {
+        llvm::consumeError(ExpectedOverlay.takeError());
+      }
+
       ASTBuildOperationRef NewBuildOp = std::make_shared<ASTBuildOperation>(
-          FileSystem, This->InvokRef, Mgr, DidFinishCallback);
+          FS, This->InvokRef, Mgr, DidFinishCallback);
```

Comment:
Shouldn’t we also use the overlay file system above in `getBuildOperationForConsumer` and `enqueueConsumer`?

Reply:
Ah, indeed. Thanks.